Disable gradient check for linalg.eig

int3 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #109359
* #109264
* __->__ #109165

Both the eager and compiled versions fail with the following message
when trying to compute the grad:

    RuntimeError: linalg_eig_backward: The eigenvectors in the complex
    case are specified up to multiplication by e^{i phi}. The specified
    loss function depends on this quantity, so it is ill-defined.

I'm not sure if there's a way to adapt the OpInfo such that the grad is
computable, but we should at least check that the forward pass is
correct.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

int3 committed:
Disable gradient check for linalg.eig

Both the eager and compiled versions fail with the following message
when trying to compute the grad:

    RuntimeError: linalg_eig_backward: The eigenvectors in the complex
    case are specified up to multiplication by e^{i phi}. The specified
    loss function depends on this quantity, so it is ill-defined.

I'm not sure if there's a way to adapt the OptInfo such that the grad is
computable, but we should at least check that the forward pass is
correct.

[ghstack-poisoned]

int3 committed:
Update on "Disable gradient check for linalg.eig"


Both the eager and compiled versions fail with the following message
when trying to compute the grad:

    RuntimeError: linalg_eig_backward: The eigenvectors in the complex
    case are specified up to multiplication by e^{i phi}. The specified
    loss function depends on this quantity, so it is ill-defined.

I'm not sure if there's a way to adapt the OptInfo such that the grad is
computable, but we should at least check that the forward pass is
correct.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

eellison commented on test/inductor/test_torchinductor_opinfo.py:
```diff
@@ -377,6 +376,8 @@ def wrapper_set_seed(op, *args, **kwargs):
     "linalg.lu_factor": {"check_gradient": False},
     "linalg.lu_factor_ex": {"check_gradient": False},
+    # grad calculation below fails for both the aten and the compiled implementation.
+    "linalg.eig": {"check_gradient": False},
```
Something is off here if we have to manually specify check_gradient.. It's weird that this is not working : https://github.com/pytorch/pytorch/blob/main/test/inductor/test_torchinductor_opinfo.py#L507-L515.

File an issue that the opinfo for linalg.eig (and friends) do not correctly specify either sample inputs or the supported_backward_dtypes ? Maybe worth mentioning that they're not currently exercised in `test_ops_gradients` according to the comment as well.

int3 replied:
Filed #109353.

> Maybe worth mentioning that they're not currently exercised in test_ops_gradients according to the comment as well.

I think the comment about `test_ops_gradients` only applies to complex32; the test is failing for the float{32,64} dtypes, I guess because float32 inputs can result in complex eigenvalues.

eellison reviewed (approved):
looks good, let's file an issue because this should be unnecessary

int3 committed:
Update on "Disable gradient check for linalg.eig"


Both the eager and compiled versions fail with the following message
when trying to compute the grad:

    RuntimeError: linalg_eig_backward: The eigenvectors in the complex
    case are specified up to multiplication by e^{i phi}. The specified
    loss function depends on this quantity, so it is ill-defined.

I'm not sure if there's a way to adapt the OpInfo such that the grad is
computable, but we should at least check that the forward pass is
correct.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

int3 committed:
Update on "Disable gradient check for linalg.eig"


Both the eager and compiled versions fail with the following message
when trying to compute the grad:

    RuntimeError: linalg_eig_backward: The eigenvectors in the complex
    case are specified up to multiplication by e^{i phi}. The specified
    loss function depends on this quantity, so it is ill-defined.

I'm not sure if there's a way to adapt the OptInfo such that the grad is
computable, but we should at least check that the forward pass is
correct.

[ghstack-poisoned]

int3 committed:
Update on "Disable gradient check for linalg.eig"


Both the eager and compiled versions fail with the following message
when trying to compute the grad:

    RuntimeError: linalg_eig_backward: The eigenvectors in the complex
    case are specified up to multiplication by e^{i phi}. The specified
    loss function depends on this quantity, so it is ill-defined.

I'm not sure if there's a way to adapt the OpInfo such that the grad is
computable, but we should at least check that the forward pass is
correct.

[ghstack-poisoned]

int3 committed:
Update on "Disable gradient check for linalg.eig"


Both the eager and compiled versions fail with the following message
when trying to compute the grad:

    RuntimeError: linalg_eig_backward: The eigenvectors in the complex
    case are specified up to multiplication by e^{i phi}. The specified
    loss function depends on this quantity, so it is ill-defined.

I'm not sure if there's a way to adapt the OpInfo such that the grad is
computable, but we should at least check that the forward pass is
correct.

[ghstack-poisoned]
